src/Serilog.Sinks.File.Archive/ArchiveHooks.cs

```diff
@@ -27,12 +30,24 @@ public class ArchiveHooks : FileLifecycleHooks
         public ArchiveHooks(CompressionLevel compressionLevel = CompressionLevel.Fastest, string targetDirectory = null)
         {
             if (compressionLevel == CompressionLevel.NoCompression && targetDirectory == null)
-                throw new ArgumentException("Either compressionLevel or targetDirectory must be set");
+                throw new ArgumentException($"Either {nameof(compressionLevel)} or {nameof(targetDirectory)} must be set");
 
             this.compressionLevel = compressionLevel;
             this.targetDirectory = targetDirectory;
         }
 
+        public ArchiveHooks(int retainedFileCountLimit, CompressionLevel compressionLevel = CompressionLevel.Fastest)
```

Comment:
Why no `targetDirectory` parameter?

Reply:
first version of this did not check if the target path was tokenised and just skipped the delete step if it was to avoid the complications of tokenised paths so I had just not included the ability to set targetDirectory, I'll add it back in with a check that it is a fixed path.